nsxt/resource_nsxt_policy_nat_rule.go

```diff
@@ -231,12 +246,15 @@ func patchNsxtPolicyNATRule(sessionContext utl.SessionContext, connector client.
 	return client.Patch(gwID, natType, *rule.Id, rule)
 }
 
-func getNatTypeByAction(action string) string {
+func getNatTypeByAction(natType string, action string) string {
 	if action == model.PolicyNatRule_ACTION_NAT64 {
 		return model.PolicyNat_NAT_TYPE_NAT64
 	}
```

Comment:
Do we still want the check above if nat_type != ""?

If nat_type is not Policy_NAT_TYPE_NAT64, The current code will "fix" an invalid configuration. It might be ok, but maybe we should let the call fail so users know they have an invalid configuration?

Reply:
Good point, I have added the validation to make sure NAT64 action can only go under NAT64 type, and vise versa.
Regarding the `natType == ""` check - this one is merely to replace default value functionality, since default can not be configured together with `Computed` marker.